### PR TITLE
Made -Subject parameter of sendMail command no longer mandatory.

### LINKED ISSUE #4043: Send-MailMessage $Subject Mandatory=$true
Not a bug, but a request for "Send-MailMessage" cmdlet

```powershell
 $From = $email
    $To = $Number
    $Body = "Something"
    $Subject = $null
    $SMTPServer = "smtp.gmail.com"
    $SMTPPort = "587"
    Send-MailMessage -From $From -to $To -Subject $Subject `
        -Body $Body -SmtpServer $SMTPServer -port $SMTPPort -UseSsl `
    -Credential $Cred
```

RESULT:
```
Send-MailMessage : Cannot validate argument on parameter 'Subject'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At line:8 char:52
+       -Body $Body -SmtpServer $SMTPServer -Subject $Subject -port $SM ...
+                                                    ~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Send-MailMessage], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Microsoft.PowerShell.Commands.SendMailMessage
```

CAUSE:
```
[Parameter(Mandatory=$true, Position=1)]
        [Alias('sub')]
        [ValidateNotNullOrEmpty()]
        [string]
        ${Subject},
```

--------
Emails do not require subjects to be present.  However, in the Send-MailMessage cmdlet, the $subject parameter is set to "Mandatory=$true".  Since an email does not require a subject, this should really be set to "$false". 

I've created my own workaround this, but I feel that this should be changed since a subject is not imperative for an email to be sent.

### PR DESCRIPTION
**Pull request by ece-jacob-scott**
Fixes https://github.com/PowerShell/PowerShell/issues/4043
- Made the subject parameter not mandatory 
- Took away null exception check for the subject

The PR just makes the subject parameter for the sendMail command able to be null. It also adds a test for the send mail function to be called with no subject parameter.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by TylerLeonhardt — APPROVED** [review]
LGTM 😊

**Review by rjmholt — APPROVED** [review]
LGTM

**Comment by TylerLeonhardt** [comment]
This is a student at [HackIllinois](https://www.hackillinois.org). @rjmholt and I are working with the students here and are reviewing PRs that they send.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1 [hunk 1]
```diff
@@ -181,4 +181,23 @@ Describe "Basic Send-MailMessage tests" -Tags CI {
         $mail.Body.Count | Should -BeExactly 1
         $mail.Body[0] | Should -BeExactly $body
     }
+
+    $ItArgs = $PesterArgs.Clone()
+    $ItArgs['Name'] = "Can send mail message from user to self without subject " + $ItArgs['Name']
+
+    It @ItArgs {
+        $body = "Greetings from me."
+        $subject = "Test message"
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1]: Please remove the line.

**Comment by iSazonov** [comment]
@adityapatwardhan Can we merge or will wait #8859?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1 (hunk 1 above)
Comment (ece-jacob-scott) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1]: Good catch I took out the subject line from the test. Thank you :).

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1 [hunk 2]
```diff
@@ -181,4 +181,23 @@ Describe "Basic Send-MailMessage tests" -Tags CI {
...
+    $ItArgs['Name'] = "Can send mail message from user to self without subject " + $ItArgs['Name']
+
+    It @ItArgs {
+        $body = "Greetings from me."
+        Send-MailMessage -To $address -From $address -ReplyTo $address -Body $body -SmtpServer 127.0.0.1
+        Test-Path -Path $mailBox | Should -BeTrue
+        $mail = read-mail $mailBox
+        $mail.From | Should -BeExactly $address
+        $mail.To.Count | Should -BeExactly 1It
+        $body = "Greetings from me."
```
Comment (TylerLeonhardt) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1]: You can delete this line I think

**Comment by iSazonov** [comment]
@ece-jacob-scott @TylerLeonhardt Could you continue and rebase to get latest tests?

**Comment by TylerLeonhardt** [comment]
@ece-jacob-scott is a master at rebasing now :) but if you need help feel free to ping me

**Comment by iSazonov** [comment]
Please rebase so we could merge.

**Review by iSazonov — APPROVED** [review]

**Review by TylerLeonhardt — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
@ece-jacob-scott Thanks for your contribution!

**Comment by TylerLeonhardt** [comment]
Wahoo! Well done! 🎉

**Comment by iSazonov** [comment]
Ah, sorry - please create new issue in PowerShell-Docs repo and add reference in the PR description.

**Pull request merged**